compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.td

```diff
@@ -22,109 +24,135 @@ def BubbleUpExpandShapesPass :
+def CollapseReductionDimensionsPass :
+    Pass<"iree-flow-collapse-reduction-dimensions", ""> {
+  let summary = "Collapse reduction dimensions when possible.";
+  let dependentDialects = [
+    "mlir::linalg::LinalgDialect",
   ];
 }
 
-def CaptureDynamicDimsPass :
-    Pass<"iree-flow-capture-dynamic-dims", ""> {
-  let summary = "Captures dynamic shape dimensions required by dispatch operands/results and control flow operations.";
+def ElementwiseOpFusionPass :
+    Pass<"iree-flow-elementwise-op-fusion", ""> {
+  let summary = "Fuse elementwise operations.";
+  let options = [
+    Option<"fuseMultiReduction", "fuse-multi-reduction", "bool",
+           /*default=*/"true", "Fuse ops that have multiple reduction iterators">
+  ];
   let dependentDialects = [
-    "IREE::Flow::FlowDialect",
+    "mlir::affine::AffineDialect",
   ];
 }
 
-def CleanupTensorShapesPass :
-    Pass<"iree-flow-cleanup-tensor-shapes", ""> {
-  let summary = "Cleans up any remaining tensor shape metadata after lowering.";
-}
-
-def CloneProducersIntoDispatchRegionsPass :
-    InterfacePass<"iree-flow-clone-producers-into-dispatch-regions", "mlir::FunctionOpInterface"> {
-  let summary = "Clone producers into dispatch regions to be isolated above.";
+def FoldUnitExtentDimsPass :
+    Pass<"iree-flow-fold-unit-extent-dims", "mlir::ModuleOp"> {
+  let summary = "Fold unit extent dimension of operations.";
   let description = [{
-    Pass to clone into dispatch regions producers of values used in the dispatch
-    regions but defined in the above. This prepares the dispatch regions for
-    converting to dispatch workgroups with explicit captures.
+    Imports upstream patterns to fold unit extent dims but with IREE control.
   }];
+  let dependentDialects = [
+    "mlir::affine::AffineDialect",
+    "mlir::arith::ArithDialect",
+    "mlir::linalg::LinalgDialect",
+    "mlir::tensor::TensorDialect",
+  ];
 }
 
-def CollapseReductionDimensionsPass :
-    Pass<"iree-flow-collapse-reduction-dimensions", ""> {
-  let summary = "Collapse reduction dimensions when possible.";
+def FusionPreprocessingPass :
+    Pass<"iree-flow-fusion-preprocessing", ""> {
+  let summary = "Run useful preprocessing patterns that help with fusion.";
   let dependentDialects = [
-    "mlir::linalg::LinalgDialect",
+    "mlir::affine::AffineDialect",
   ];
 }
 
-def ConvertMeshToFlowPass :
-    Pass<"iree-convert-mesh-to-flow", "mlir::ModuleOp"> {
-  let summary = "Convert Mesh dialect operations to flow.";
-  let description = [{
-    Each mesh corresponds to a default !flow.channel with the same group name.
-    ```
-    mesh.mesh @mesh_1(shape = 2x3)
-    ```
-    ```
-    %channel = flow.channel.default "mesh_1" : !flow.channel
-    ```
-    If there is onl one mesh in the program than the name is omitted and the
-    ```
-    %channel = flow.channel.default : !flow.channel
-    ```
+def FuseMultiUseElementwiseProducerPass :
```

Comment:
nit: `FuseMultiUse...` should come before `FusionPreprocessing` in this section.

Reply:
I am going to practice "A... B... C... D... E.. F.. G.." with my 4 year old :)